docs/DeveloperGuide.md

```diff
@@ -976,10 +1013,128 @@ Priorities: High - (must have), Medium - (nice to have), Low -  (unlikely to hav
 **Extensions**
 
 * 2a. BTBB detects an error in the entered data.
-    * 2a1. BTBB informs user why marking an order as undone failed.
+    * 2a1. BTBB informs Home Chef why marking an order as undone failed.
+
+      Use case ends.
+
+**Use case: UC18 Add recipe**
+
+**MSS**
+
+1. Home chef chooses to add a recipe.
+2. Home chef enters details in required format.
+2. BTBB displays new recipe.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. Recipe entered already exists in the system.
+    * 2a1. BTBB informs Home Chef that recipe already exists.
+
+      Use Case ends.
+
+* 2b. BTBB detects an error in the entered data.
+ 	* 2b1. BTBB informs Home Chef why addition of recipe did not happen.
+
+      Use case ends.
+
+**Use case: UC19 Find recipe**
+
+**MSS**
+
+1. Home chef chooses to find a recipe by a field.
+2. Home chef enters search terms in required format.
+3. BTBB displays all recipes that match the terms.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. BTBB detects an error in the entered data.
+    * 2a1. BTBB informs Home Chef why finding a recipe failed.
+
+      Use Case ends.
+
+**Use case: UC20 Delete recipe**
+
+**MSS**
+
+1. Home chef chooses to delete a recipe.
+2. Home chef enters details in required format.
+3. BTBB deletes recipe.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. BTBB detects an error in the entered data.
+    * 2a1. BTBB informs Home Chef why deletion did not happen.
+
+      Use Case ends.
+
+**Use case: UC21 View all recipes**
+
+**MSS**
+
+1. Home chef chooses to view all recipes.
+2. Home chef enters command to see all recipes.
+3. BTBB displays all recipes.
+
+   Use case ends.
+
+**Use case: UC22 Edit recipe**
+
+**MSS**
+
+1. Home chef chooses to edit a recipe.
+2. Home chef enters details in required format.
+3. BTBB displays the edited recipe.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. Recipe entered already exists in the system.
+    * 2a1. BTBB informs Home Chef that recipe already exists.
 
       Use case ends.
 
+* 2b. BTBB detects an error in the entered data.
+    * 2b1. BTBB informs Home Chef why editing of recipe failed.
+
+      Use case ends.
+
+**Use case: UC23 View revenue for each month for past 12 months**
+
+**MSS**
+
+1. Home chef wants to view revenue earned for each month for past 12 months.
+2. Home chef switches to statistics tab.
+3. BTBB displays revenue for each month for the past 12 months.
+
+   Use case ends.
+
```

Comment:
Should we have an extension for tab command failure or do you think it's unnecessary?

Reply:
I don't think its necessary